lib/puppet/provider/netscaler_file/rest.rb

```diff
@@ -0,0 +1,58 @@
+require 'puppet/provider/netscaler'
+require 'base64'
+require 'json'
+
+Puppet::Type.type(:netscaler_file).provide(:rest, parent: Puppet::Provider::Netscaler) do
+  def netscaler_api_type
+    "systemfile"
+  end
+
+  def self.instances
+    instances = []
+    #look for files at a certain location
+    files = Puppet::Provider::Netscaler.call('/config/systemfile',{'args'=>"filelocation:%2Fnsconfig%2F"})
+    return [] if files.nil?
+
+    files.each do |file|
+      binds = Puppet::Provider::Netscaler.call("/config/systemfile", {'args'=>"filelocation:%2Fnsconfig%2F,filename:#{file['filename']}"}) || [] 
+      binds.each do |bind|
+        instances << new(
+          :ensure       => :present,
+          :name         => bind['filename'],
+          :filecontent  => bind['filecontent'],
```

Comment:
Either `file_content` or just `content` would probably be more in line with our style.